### PR TITLE
fix(eslint): better messaging and bug fix (#3950

### PR DESCRIPTION
**Pull request by tknickman**
A few fixes for the eslint plugin. 

1. Fix error message to remove the leading `$` since that was deprecated in 1.5
2. Provides more explicit error messaging 
3. Fixes a bug that could cause the rule to exit early if no global deps were provided
4. Cleans up / standardizes tests and mocks

This should also fix https://github.com/vercel/turbo/issues/3943

### CODE REVIEW

**Comment by tknickman** [comment]
Published this as a canary for a few discord members to try out: https://www.npmjs.com/package/eslint-plugin-turbo/v/0.0.9-canary.0

File: packages/eslint-plugin-turbo/lib/rules/no-undeclared-env-vars.ts [hunk 1]
```diff
@@ -76,12 +77,20 @@ function create(context: Rule.RuleContext): Rule.RuleListener {
...
   const filePath = getPhysicalFilename();
-  const allTurboVars =
-    getEnvVarDependencies({
-      cwd,
-    }) || {};
+  const allTurboVars = getEnvVarDependencies({
+    cwd,
+  });
+
+  // if this returns null, something went wrong reading from the turbo config
```
Comment (nathanhammond) [inline packages/eslint-plugin-turbo/lib/rules/no-undeclared-env-vars.ts:84]:
```suggestion
  // if allTurboVars is null, something went wrong reading from the turbo config
```

File: packages/eslint-plugin-turbo/__fixtures__/configs/single/turbo.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (nathanhammond) [inline packages/eslint-plugin-turbo/__fixtures__/configs/single/turbo.json:1]: Needs more comments. 😜

**Review by nathanhammond — APPROVED** [review]
🛠🚢

**Comment by wladiston** [comment]
Not sure if the benchmark is testing it but the `turbo/no-undeclared-env-vars` rule seems to be extremely slow on the `0.0.9`. 
Confirming it does work with the new turbo.json under the project's folder 👍 
Screenshot 2023-02-27 at 3 44 02 pm

**Pull request merged**